src/wp-includes/rest-api/endpoints/class-wp-rest-autosaves-controller.php

```diff
@@ -234,8 +234,8 @@ public function create_item( $request ) {
 			 */
 			$autosave_id = wp_update_post( wp_slash( (array) $prepared_post ), true );
 		} else {
-			// Non-draft posts: create or update the post autosave.
-			$autosave_id = $this->create_post_autosave( (array) $prepared_post );
+			// Non-draft posts: create or update the post autosave. Pass the meta data.
+			$autosave_id = $this->create_post_autosave( (array) $prepared_post, $request->get_param( 'meta' ) );
```

Comment:
```suggestion
			$autosave_id = $this->create_post_autosave( (array) $prepared_post, (array) $request->get_param( 'meta' ) );
```

@adamsilverstein that should fix the unit tests.